hokusai/services/deployment.py

```diff
@@ -79,8 +79,8 @@ def update(self, tag, constraint, git_remote, resolve_tag_sha1=True):
       remote = git_remote or config.git_remote
       if remote is not None:
         print_green("Pushing deployment tags to %s..." % remote)
-        shout("git tag -f %s" % self.context, print_output=True)
-        shout("git tag -f %s" % deployment_tag, print_output=True)
+        shout("git tag -f %s %s" % (self.context, tag), print_output=True)
+        shout("git tag -f %s %s" % (deployment_tag, tag), print_output=True)
```

Comment:
Yikes, sounds like a tricky one to debug 😄

Reply:
Yeah - so this is a pretty gnarly little bug... overlooking passing the target ref as the second argument to `git tag` means that we've been mis-matching tags pushed to git on `hokusai deploy` / `hokusai pipeline promote` commands, as they tagged wherever `HEAD` happened to be rather than the correct tag being deployed :/

Reply:
Yikes, yeah. We fixed a very similar bug on our iOS developments just this week: https://github.com/artsy/eigen/pull/2807

Reply:
One caveat here - this is going to fail after the deployment succeeds, but while creating the git tags if your local git repo does not contain the SHA1 being deployed / promoted...  i.e. you don't have an up-to-date `upstream/master`.

Try and create the deployment tag but do not push it _before_ performing the deploy / promote - push it after a successful rollout.

@alloy @dblandin wdyt, as you ran into problems like this when promoting MP...

Reply:
Another option would be to just issue a warning if git tags could not be created / pushed for any reason.

Reply:
We already know the remote via the `--git-remote` flag. Would it be valuable to run a `git fetch` to pull down the sha we're deploying?